### PR TITLE
Remove nl.c from libmonodroid.so builds

### PR DESCRIPTION
**Pull request by CoffeeFlux**
A bug in mono required changing the signature of CreateNLSocket and CloseNLSocket, and now that those are icalls on Android this is the easiest way to enable changing the signature.

### CODE REVIEW

File: src/monodroid/jni/config.h [hunk 1]
```diff
@@ -1,11 +0,0 @@
-#ifndef INC_MONODROID_CONFIG_H
```
Comment (grendello) [inline src/monodroid/jni/config.h]: Removal of `config.h` breaks the build, please fix.

Linux:
```
[2019-06-25T19:53:57.204Z]   /mnt/jenkins/workspace/xamarin-android-linux-pr-pipeline/xamarin-android/external/mono/support/zlib-helper.c:9:10: fatal error: 'config.h' file not found
[2019-06-25T19:53:57.204Z]   #include <config.h>
[2019-06-25T19:53:57.204Z]            ^~~~~~~~~~
[2019-06-25T19:53:57.204Z]   1 error generated.
```

Mac:
```
[2019-06-25T20:57:20.844Z]   /Users/builder/jenkins/workspace/xamarin-android-pr-pipeline-release/xamarin-android/external/mono/support/zlib-helper.c:9:10: fatal error: 'config.h' file not found
[2019-06-25T20:57:20.844Z]   #include <config.h>
[2019-06-25T20:57:20.844Z]            ^~~~~~~~~~
[2019-06-25T20:57:20.844Z]   1 error generated.
```

**Comment by CoffeeFlux** [comment]
I'm not actually sure why that file is including config.h - looking through, it doesn't seem to reference anything in the config. Even if I changed that in upstream mono though, I suppose it wouldn't actually propagate down for quite a while. Do you want me to just have the build system generate an empty config.h?

File: src/monodroid/monodroid.targets [hunk 2]
```diff
@@ -15,13 +15,7 @@
     <_EmbeddedBlobDestination Include="jni\machine.config.include" />
   </ItemGroup>
   <Target Name="_BuildRuntimes"
-      DependsOnTargets="_GenerateIncludeFiles;_BuildAndroidRuntimes;_BuildHostRuntimes">
-  </Target>
-  <Target Name="_GenerateIncludeFiles"
```
Comment (jonpryor) [inline src/monodroid/monodroid.targets]: Why are you removing this target?

Removing `support/nl.c` does not in any way change our need for `monodroid_config` or `monodroid_machine_config`, both of which are generated by `<GenerateMonoDroidIncludes/>`; see also:

https://github.com/xamarin/xamarin-android/blob/60c1ee42437066366b4da5f94ad47d41dcb26284/src/monodroid/jni/monodroid-glue.cc#L2009-L2010

DO NOT remove the `_GenerateIncludeFiles` target.  You only need to update the `Inputs` so that `jni\config.h` isn't specified.

***Or...***, leave this entirely unchanged, and "revert" the `config.h` removal (and just remove the declarations of `CreateNLSocket()`/etc.), so that the `#include <config.h>` usage continues to work.

Comment (CoffeeFlux) [reply src/monodroid/monodroid.targets]: Ah okay my bad, I misread what you'd said on Slack. Leaving what's just basically an empty config file seems gross, but I guess that's the easiest option for now...

File: src/monodroid/jni/config.h [hunk 3]
```diff
@@ -4,8 +4,6 @@
 #include <monodroid.h>
 #include <stdint.h>
 
-MONO_API  int CreateNLSocket (void);
-MONO_API  int ReadEvents (void *sock, void *buffer, int32_t count, int32_t size);
-MONO_API  int CloseNLSocket (void *sock);
+// This file only exists for the include in zlib.c
```
Comment (lambdageek) [inline src/monodroid/jni/config.h]: This kind of comment has a habit of becoming outdated with alarming frequency.

File: src/monodroid/monodroid.targets [hunk 4]
```diff
@@ -14,9 +14,11 @@
     <_EmbeddedBlobSource Include="machine.config.xml" />
     <_EmbeddedBlobDestination Include="jni\machine.config.include" />
   </ItemGroup>
+
```
Comment (lambdageek) [inline src/monodroid/monodroid.targets]: unrelated cosmetic change

File: src/monodroid/monodroid.targets [hunk 5]
```diff
@@ -14,9 +14,11 @@
...
   <Target Name="_BuildRuntimes"
       DependsOnTargets="_GenerateIncludeFiles;_BuildAndroidRuntimes;_BuildHostRuntimes">
   </Target>
+
```
Comment (lambdageek) [inline src/monodroid/monodroid.targets]: unrelated cosmetic change

File: src/monodroid/jni/config.h (hunk 3 above)
Comment (CoffeeFlux) [reply src/monodroid/jni/config.h]: Good point. I didn't think about it since I intend to get rid of the include mandating its presence, but it's probably worth making more generic for now.

File: src/monodroid/jni/config.h [hunk 6]
```diff
@@ -4,8 +4,6 @@
 #include <monodroid.h>
 #include <stdint.h>
 
-MONO_API  int CreateNLSocket (void);
-MONO_API  int ReadEvents (void *sock, void *buffer, int32_t count, int32_t size);
-MONO_API  int CloseNLSocket (void *sock);
+// This file currently does nothing, but is needed for builds to succeed.
```
Comment (jaykrell) [inline src/monodroid/jni/config.h]: Incorrect comment -- it isn't doing nothing, it has `#include`.

Comment (CoffeeFlux) [reply src/monodroid/jni/config.h]: Fair point.
